sunpy/coordinates/tests/test_transformations.py

```diff
@@ -5,10 +5,49 @@
 from astropy.coordinates import SkyCoord, get_body_barycentric
 from astropy.time import Time
 
-from sunpy.coordinates import Helioprojective, HeliographicStonyhurst, HeliographicCarrington, get_sun_L0
+import sunpy.coordinates.transformations as trans
+from sunpy.coordinates import (Helioprojective, HeliographicStonyhurst,
+                               HeliographicCarrington, get_sun_L0,
+                               Heliocentric)
 from sunpy.time import parse_time
 
 
+def test_new_hcc_to_hgs():
+    # Generate 10 random coordinates
+    for i in range(10):
+        x = np.random.rand() * u.km
+        y = np.random.rand() * u.km
+        z = np.random.rand() * u.km
+        time = '2007-05-04T21:08:12'
+        hcc = Heliocentric(x=x, y=y, z=z, obstime=time)
+        hgs_frame = HeliographicCarrington(obstime=time)
+
+        # New matrix based transformation
+        out1 = hcc.transform_to(HeliographicStonyhurst)
+        # Old function based transformation
+        out2 = trans.old_hcc_to_hgs(hcc, hgs_frame)
+        # Check that they give the same results
+        assert_quantity_allclose(out1.lon, out2.lon)
+        assert_quantity_allclose(out1.lat, out2.lat)
+        assert_quantity_allclose(out1.radius, out2.radius)
+
+
+def test_hcc_to_hgs():
+    '''
+    Check that a coordinate pointing to the observer in Heliocentric
+    coordinates maps to the lattitude/longitude of the observer in
+    HeliographicStonyhurst coordinates.
+    '''
+    lat = 10 * u.deg
+    lon = 20 * u.deg
+    observer = HeliographicStonyhurst(lat=lat, lon=lon)
+    hcc_in = Heliocentric(x=0*u.km, y=0*u.km, z=1*u.km, observer=observer)
+    hgs_out = hcc_in.transform_to(HeliographicStonyhurst)
+
+    assert_quantity_allclose(hgs_out.lat, lat)
+    assert_quantity_allclose(hgs_out.lon, lon)
```

Comment:
Eeeek adding this test has revealed that `hcc_to_hgs` (even the old version) is broken! It doesn't look like the transformation code ever takes into account the observer longitude.

Reply:
The current version of hcc_to_hgs appears to include the observer longitude

https://github.com/sunpy/sunpy/blob/master/sunpy/coordinates/transformations.py#L159
https://github.com/sunpy/sunpy/blob/master/sunpy/coordinates/transformations.py#L166

Is this code incorrect?  It looks OK compared to Thompson (2006), A&A 449, 791–803, eq. 11.

Reply:
Woops, I must have missed that when I was writing this originally. I'll try and add it in (and hopefully the test will pass)